From 56c36f67af25443870773633f6e68beccdf436e5 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Sun, 25 Oct 2026 13:16:06 +0100
Subject: [PATCH] Fix compilation error for implicit declaration of memcpy

memcpy require the string.h header or a compilation error is triggered:

allocate.c: In function 'debugRealloc':
allocate.c:349:17: error: implicit declaration of function 'memcpy' [-Wimplicit-function-declaration]
  349 |                 memcpy(result, memoryParameter, copyLength);
      |                 ^~~~~~
allocate.c:29:1: note: include '<string.h>' or provide a declaration of 'memcpy'
   28 | #include "allocate.h"
  +++ |+#include <string.h>
   29 |
allocate.c:349:17: warning: incompatible implicit declaration of built-in function 'memcpy' [-Wbuiltin-declaration-mismatch]
  349 |                 memcpy(result, memoryParameter, copyLength);
      |                 ^~~~~~
allocate.c:349:17: note: include '<string.h>' or provide a declaration of 'memcpy'

Add the missing header to fix the compilation error.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 allocate.c | 1 +
 1 file changed, 1 insertion(+)

--- a/allocate.c
+++ b/allocate.c
@@ -23,6 +23,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 #include "os.h"
 #include "allocate.h"
